### ISSUE TITLE
Potential bug of GAN example

### ISSUE THREAD

**Issue by dolaameng** [yunjey/pytorch-tutorial#56]
Hi,

First thank you for sharing the tutorials! 

I have a quick question on the [GAN example](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/02-intermediate/generative_adversarial_network/main.py#L78), where  'detaching' the variable should be used in my opinion like `fake_images = G(z).detach()`, to avoid the training of Discriminator influences the Generator's behaviour at this stage.  Or you have specific considerations here?

Thank you!

**Comment by yunjey** [comment]
@dolaameng No, it doesn't matter because we only train the discriminator using [d_optimizer.step()](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/03-advanced/deep_convolutional_gan/solver.py#L96) and reset the  [grad buffer](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/02-intermediate/generative_adversarial_network/main.py#L97-L98) before calling `g_loss.backward()`. In the case of `fake_images =Variable(G(z).detach().cuda())`, it saves memory and speed up the training, but in my experience this is trivial.

**Comment by dolaameng** [comment]
Hi @yunjey yeah it makes sense. Thank you for clarification.
